load.go

```diff
@@ -447,3 +453,57 @@ func (b *ArtifactBuild) processBuildArgs(lex *shell.Lex, args map[string]string,
 
 	return goerrors.Join(errs...)
 }
+
+func (img *ImageConfig) processBuildArgs(lex *shell.Lex, args map[string]string, allowArg func(string) bool) error {
+	var errs error
+	for _, p := range []*string{&img.Base, &img.Cmd, &img.User, &img.Entrypoint, &img.StopSignal, &img.WorkingDir} {
+		updated, err := expandArgs(lex, *p, args, allowArg)
+		if err != nil {
+			errs = goerrors.Join(errs, errors.Wrap(err, "imgconfig"))
+		}
+		*p = updated
+	}
+
+	for i, s := range img.Env {
+		updated, err := expandArgs(lex, s, args, allowArg)
+		if err != nil {
+			errs = goerrors.Join(errs, errors.Wrapf(err, "env %s", s))
+			continue
+		}
+		img.Env[i] = updated
+	}
+
+	for k, v := range img.Labels {
+		updated, err := expandArgs(lex, v, args, allowArg)
+		if err != nil {
+			errs = goerrors.Join(errs, errors.Wrapf(err, "env %s=%s", k, v))
+			continue
+		}
+		img.Labels[k] = updated
+	}
+
+	if img.Post != nil {
```

Comment:
I don't see why this would be necessary and would prefer to not support it.

Reply:
Understandable. Not strictly necessary, just a little more convenient for certain things. If you don't agree with the rationale feel free to close